lib/memo_wise.rb

```diff
@@ -423,6 +423,28 @@ def #{method_name}#{args_str}
           target.singleton_class.send(:private, method_name)
         end
       end
+
+      # Override [Module#instance_method](https://ruby-doc.org/core-3.0.0/Module.html#method-i-instance_method)
+      # to proxy the original `UnboundMethod#parameters` results. This proxying
+      # helps code that uses Ruby reflection over method parameters
+      def target.instance_method(symbol)
+        original_memo_wised_name = :"_memo_wise_original_#{symbol}"
+        curr_method = super
+
+        orig_method =
+          if symbol == :initialize
+            curr_method.super_method
+          elsif private_method_defined?(original_memo_wised_name)
+            super(original_memo_wised_name)
+          end
+
+        unless orig_method.nil?
+          orig_params = orig_method.parameters
+          curr_method.define_singleton_method(:parameters) { orig_params }
```

Comment:
This approach makes sense to me, but I just want to make sure we're on the same page that we think this is better than just proxying the whole `UnboundMethod` object instead.

Reply:
Discussed live - I'm open on this - was leaning towards minimal "lying" to start to avoid unintentionally breaking things

Reply:
Agreed to go with just proxying the original parameter values, for minimally-intrusive change